src/policy/rbf.cpp

```diff
@@ -174,3 +167,35 @@ std::optional<std::string> PaysForRBF(CAmount original_fees,
     }
     return std::nullopt;
 }
+
+std::optional<std::string> CheckAncestorScores(CAmount replacement_fees,
```

Comment:
In this function, we should be thinking about the "ancestor score" of a transaction as the minimum of its ancestor feerate and its own feerate.  A transaction should never sort higher in the ancestor scoring comparison than it's own feerate, because the parent transactions can be included without the child.